### PR TITLE
javascript: Add support for changed include path in emscripten 1.38.42

### PR DESCRIPTION
**Pull request by andrewleech**
One of the system include paths in emscripten changed in 1.38.42 which breaks the clang precompiling stage of the javascript port make process.

This change simply adds the extra include path as needed, leaving the previous ones in place so as to not break build with existing installations.

For further discussion see https://github.com/trzecieu/emscripten-docker/issues/51

### CODE REVIEW

**Comment by dpgeorge** [comment]
Thanks, it looks like it will clash a bit with #4875, but is kind of independent to that.  Anyway, the patch here is small and understandable.

**Comment by pmp-p** [comment]
i suggested in https://github.com/micropython/micropython/pull/4875 to open an issue about the preprocessor  behaviour of emcc here https://github.com/emscripten-core/emscripten/issues since the problem with multiple files is your findings and you may know better how to pack a reproducer.  


OFF TOPIC:

> the patch here is small and understandable.

unlike someone's build system.

if patch needed to be splitted or documented then https://github.com/micropython/micropython/pull/4875 was the place to say it or : please close  to your liking instead of letting me - wasting my time - adding tested support for native clang, all emsdk flavours and user mods.

**Comment by andrewleech** [comment]
@pmp-p  I was unaware of your pr when I ran into this issue, this is a minor update to workaround this single issue I ran into while putting together a multi-port CI proof of concept.

Your PR currently has a lot of fixes/features in it, which generally look like positive moves forward for the the javascript port! 
I also agree it would be better to use emcc for the preprocessing too, but if you're going to need a long workaround for a bug in emcc then that's going to be a maintenance burden in the future .

I would suggest that the other changes in your PR that aren't directly related to preprocessing would be easier to review and merge if they were broken out into clean commits/pr's because the currently 26 commits there are a bit hard to follow.

**Comment by pmp-p** [comment]
> I also agree it would be better to use emcc for the preprocessing too.

Looking at the number of declinaison of emsdk portable flavours and system wide pre installed emscripten : it's the only safe way.

given the problem emsdk people generally have to solve i think the preprocessing bug ( if any , there's sometimes holes in documentation ) will look trivial and understandable.

but because they are very busy but also very reactive people: they just need a small repro case.

i have no idea what @dpgeorge did to make emcc -E fail at the time since he did not provide a testcase to work on.

**Comment by andrewleech** [comment]
I just tried to replace clang with emcc for the preprocessing and came up with this error
```
docker run --rm -it -v `pwd`:/src trzeci/emscripten:sdk-tag-1.38.43-64bit make -C /src/ports/javascript V=1
make: Entering directory '/src/ports/javascript'
python3 ../../py/makeversionhdr.py build/genhdr/mpversion.h
GEN build/genhdr/qstr.i.last
emcc -E -DNO_QSTR -Ibuild/tmp -m32 -Wall -Werror -I. -I../.. -Ibuild -std=c99  -O0 -DNDEBUG -fdata-sections -ffunction-sections ../../py/mpstate.c ../../py/malloc.c ../../py/gc.c ../../py/pystack.c ../../py/qstr.c ../../py/vstr.c ../../py/mpprint.c ../../py/unicode.c ../../py/mpz.c ../../py/reader.c ../../py/lexer.c ../../py/parse.c ../../py/scope.c ../../py/compile.c ../../py/emitcommon.c ../../py/emitbc.c ../../py/asmbase.c ../../py/asmx64.c ../../py/emitnx64.c ../../py/asmx86.c ../../py/emitnx86.c ../../py/asmthumb.c ../../py/emitnthumb.c ../../py/emitinlinethumb.c ../../py/asmarm.c ../../py/emitnarm.c ../../py/asmxtensa.c ../../py/emitnxtensa.c ../../py/emitinlinextensa.c ../../py/formatfloat.c ../../py/parsenumbase.c ../../py/parsenum.c ../../py/emitglue.c ../../py/persistentcode.c ../../py/runtime.c ../../py/runtime_utils.c ../../py/scheduler.c ../../py/nativeglue.c ../../py/stackctrl.c ../../py/argcheck.c ../../py/warning.c ../../py/map.c ../../py/obj.c ../../py/objarray.c ../../py/objattrtuple.c ../../py/objbool.c ../../py/objboundmeth.c ../../py/objcell.c ../../py/objclosure.c ../../py/objcomplex.c ../../py/objdeque.c ../../py/objdict.c ../../py/objenumerate.c ../../py/objexcept.c ../../py/objfilter.c ../../py/objfloat.c ../../py/objfun.c ../../py/objgenerator.c ../../py/objgetitemiter.c ../../py/objint.c ../../py/objint_longlong.c ../../py/objint_mpz.c ../../py/objlist.c ../../py/objmap.c ../../py/objmodule.c ../../py/objobject.c ../../py/objpolyiter.c ../../py/objproperty.c ../../py/objnone.c ../../py/objnamedtuple.c ../../py/objrange.c ../../py/objreversed.c ../../py/objset.c ../../py/objsingleton.c ../../py/objslice.c ../../py/objstr.c ../../py/objstrunicode.c ../../py/objstringio.c ../../py/objtuple.c ../../py/objtype.c ../../py/objzip.c ../../py/opmethods.c ../../py/sequence.c ../../py/stream.c ../../py/binary.c ../../py/builtinimport.c ../../py/builtinevex.c ../../py/builtinhelp.c ../../py/modarray.c ../../py/modbuiltins.c ../../py/modcollections.c ../../py/modgc.c ../../py/modio.c ../../py/modmath.c ../../py/modcmath.c ../../py/modmicropython.c ../../py/modstruct.c ../../py/modsys.c ../../py/moduerrno.c ../../py/modthread.c ../../py/vm.c ../../py/bc.c ../../py/showbc.c ../../py/repl.c ../../py/smallint.c ../../py/frozenmod.c ../../extmod/moductypes.c ../../extmod/modujson.c ../../extmod/modure.c ../../extmod/moduzlib.c ../../extmod/moduheapq.c ../../extmod/modutimeq.c ../../extmod/moduhashlib.c ../../extmod/moducryptolib.c ../../extmod/modubinascii.c ../../extmod/virtpin.c ../../extmod/machine_mem.c ../../extmod/machine_pinbase.c ../../extmod/machine_signal.c ../../extmod/machine_pulse.c ../../extmod/machine_i2c.c ../../extmod/machine_spi.c ../../extmod/modussl_axtls.c ../../extmod/modussl_mbedtls.c ../../extmod/modurandom.c ../../extmod/moduselect.c ../../extmod/moduwebsocket.c ../../extmod/modwebrepl.c ../../extmod/modframebuf.c ../../extmod/vfs.c ../../extmod/vfs_reader.c ../../extmod/vfs_posix.c ../../extmod/vfs_posix_file.c ../../extmod/vfs_fat.c ../../extmod/vfs_fat_diskio.c ../../extmod/vfs_fat_file.c ../../extmod/utime_mphal.c ../../extmod/uos_dupterm.c ../../lib/embed/abort_.c ../../lib/utils/printf.c build/genhdr/moduledefs.h main.c mphalport.c modutime.c ../../py/mpconfig.h mpconfigport.h >build/genhdr/qstr.i.last
Traceback (most recent call last):
  File "/emsdk_portable/emscripten/tag-1.38.43/emcc.py", line 3556, in <module>
    sys.exit(run(sys.argv))
  File "/emsdk_portable/emscripten/tag-1.38.43/emcc.py", line 1774, in run
    assert header.endswith(HEADER_ENDINGS), 'if you have one header input, we assume you want to precompile headers, and cannot have source files or other inputs as well
```

**Comment by andrewleech** [comment]
Ok, I've raised an issue with emscripten: https://github.com/emscripten-core/emscripten/issues/9466

**Comment by pmp-p** [comment]
>  I was unaware of your pr when I ran into this issue

*that* is very understandable so i tried to gather everything that can relate to fix the port here #4993 to get a synthetic view. please feel free to add anything you could think of.

**Comment by dpgeorge** [comment]
The build issue should be fixed by c13853f4dab9c65ee6d1af99b25f999b16ef2827
